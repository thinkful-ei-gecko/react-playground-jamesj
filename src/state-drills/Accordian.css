ul {
  padding: 1em;
  list-style-type: none;
  display: column;
}

li {
  border: 2px solid grey;
  background-color: black;
}

p {
  background-color: rgb(75, 73, 73);
  color: white;
  padding: 1em;
  margin: 0;
}

button {
  width: 10%;
  padding: 2em;
  border: none;
  border-bottom: 1px solid white;
  border-right: 1px solid white;
  background-color: rgb(75, 73, 73);
  color: white;
}

button:focus {
  outline: none;
}