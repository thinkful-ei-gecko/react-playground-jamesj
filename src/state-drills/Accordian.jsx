import React from 'react';

export default class Accordian extends React.Component {

  static defaultProps = {
    sections: []
  }

  state = {
    currentMenuIndex: null
  }

  handleButtonClick = (index) => {
    this.setState({
      currentMenuIndex: index
    })
  }

  renderMenuIndexContent = () => {
    return (
      <p>
        {this.state.currentMenuIndex !== null && this.props.sections[this.state.currentMenuIndex].content}
      </p>
    )
  }

  renderMenuItems() {
    const menu = this.props.sections.map((item, index) => (
      <li key={index}>
        <button key={index} onClick={() => this.handleButtonClick(index)}>{item.title}</button>
        {this.state.currentMenuIndex === index && this.renderMenuIndexContent()}
      </li>
    ))
    return menu
  }

  render() {
    return (
      <ul>
        {this.renderMenuItems()}
      </ul>
    )
  }
}