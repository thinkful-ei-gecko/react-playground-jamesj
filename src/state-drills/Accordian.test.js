import React from 'react';
import ReactDOM from 'react-dom';
import Accordian from './Accordian';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Accordian component', () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ];

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders an empty menu item when no value given', () => {
    const wrapper = shallow(<Accordian />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders no sections active by default', () => {
    const wrapper = shallow(<Accordian sections={sections} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('opens a clicked section', () => {
    const wrapper = shallow(<Accordian sections={sections} />);
    wrapper.find('button').at(2).simulate('click');
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('only keeps latest section open when multiple are clicked', () => {
    const wrapper = shallow(<Accordian sections={sections} />);
    wrapper.find('button').at(2).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});