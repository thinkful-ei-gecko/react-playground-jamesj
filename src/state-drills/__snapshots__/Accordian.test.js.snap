// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian component only keeps latest section open when multiple are clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordian component renders an empty menu item when no value given 1`] = `<ul />`;

exports[`Accordian component renders no sections active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
